tests/NexusMods.UI.Tests/Helpers/DoubleClickHelperTests.cs

```diff
@@ -0,0 +1,53 @@
+using Avalonia.Automation.Peers;
+using Avalonia.Controls;
+using Avalonia.Interactivity;
+using FluentAssertions;
+using NexusMods.App.UI.Helpers;
+
+namespace NexusMods.UI.Tests.Helpers;
+
+public class DoubleClickHelperTests : AUiTest
+{
+    public DoubleClickHelperTests(IServiceProvider provider) : base(provider) { }
+
+    [Fact]
+    public async Task Test_DoubleClick()
+    {
+        await OnUi(() =>
+        {
+            var control = new Button();
+            var automationPeer = new ButtonAutomationPeer(control);
+
+            var numPressed = 0;
+            var numTriggered = 0;
+            using var disposable1 = control.AddDisposableHandler(Button.ClickEvent, (_, _) =>
+            {
+                numPressed++;
+            }, routes: RoutingStrategies.Bubble, handledEventsToo: true);
+            using var disposable2 = control.AddDoubleClickHandler(Button.ClickEvent).Subscribe(_ => numTriggered += 1);
+
+            numPressed.Should().Be(0);
+            numTriggered.Should().Be(0);
+
+            automationPeer.Invoke();
+
+            numPressed.Should().Be(1);
+            numTriggered.Should().Be(0, because: "button was clicked once, it wasn't a double-click");
+
+            Sleep();
+
+            automationPeer.Invoke();
+
+            numPressed.Should().Be(2);
+            numTriggered.Should().Be(0, because: "button was clicked twice, but the time difference was too great");
+
+            automationPeer.Invoke();
+
+            numPressed.Should().Be(3);
+            numTriggered.Should().Be(1, because: "button was double-clicked");
+        });
+
+        return;
+        static void Sleep() => Thread.Sleep(DoubleClickHelper.DefaultTimeout + TimeSpan.FromMilliseconds(10));
```

Comment:
I would use `Task.Delay()` instead of `Thread.Sleep`. Add async in `OnUi async () =>` at the top and await it.

Reply:
The code inside `OnUi` is supposed to run synchronously to prevent weirdness.

Reply:
From my understanding the async version should work fine, I also tested it locally. The continuation after the `await Sleep()` call should be scheduled on the starting thread, because we don't use `ConfigureAwait(false)`, and the starting thread should be the UI one thanks to the dispatcher inside the `OnUI(Func<Task> action)` overload of `OnUI`.

Reply:
I don't really see any benefit in doing this. Adding async isn't really an improvement.